cni/network/network.go

```diff
@@ -1024,8 +1024,12 @@ func (plugin *NetPlugin) Delete(args *cniSkel.CmdArgs) error {
 	telemetry.LogAndSendEvent(plugin.tb, fmt.Sprintf("Deleting endpoint:%v", endpointID))
 	// Delete the endpoint.
 	if err = plugin.nm.DeleteEndpoint(cnsclient, networkID, endpointID); err != nil {
-		err = plugin.Errorf("Failed to delete endpoint: %v", err)
-		return err
+		pluginErr := plugin.RetriableError(fmt.Errorf("failed to delete endpoint: %w", err))
+		// If it is not a networkNotFound error, then return a retriable error so the container runtime will keep retrying.
+		// If it is a networkNotFoundError, then simply skip it and move on to cleanup of InfraVnet.
+		if !network.IsNetworkNotFoundError(err) {
```

Comment:
isn't it endpoint not found?

Reply:
we already skip endpoint not found by returning a nil error if the endpoint is not found, but before that we call `getNetwork` which could return `errNetworkNotFound`. If that happens, we also want to skip it since that means there is no endpoint if there is no network. Thus we can move on to the infra vnet cleanup. If we get any other error, we want to retry later.

Reply:
if network not found, we return nil too above (line no: 979)..probably we dont need to check this at all IsNetworkNotFoundError

Reply:
one more thing to verify is that are we handling if hns delete endpoint call returns endpoint not found error.. in that case we shouldn't return error

Reply:
we first check if endpoint is found before hns call so we shouldn't end up in that scenario, and even so if we do, then the next time the cni is invoked we would hit that check and it would succeed